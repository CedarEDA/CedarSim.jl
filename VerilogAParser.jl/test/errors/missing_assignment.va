module missing_assignment(p, n);

    analog function integer PAeffGEO;
        input nf, geo, minSD, Weffcj, DMCG, DMCI, DMDG;
        output Ps, Pd, As, Ad;

        integer geo, minSD;
        real Weffcj, DMCG, DMCI, DMDG, Ps, Pd, As, Ad;

        begin
            if (geo < 9)
                NumFingerDiff(nf, minSD, niIntD, nuEndD, nuIntS, nuEndS);
        end
    endfunction

    inout p, n;
    electrical p, n;

    analog begin
        I(p,n) <+ 0;
    end
endmodule
