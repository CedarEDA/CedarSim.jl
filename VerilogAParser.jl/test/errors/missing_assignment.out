[31m[1m╭ ERROR [VAS14]:[22m[39m[31m Statements without assignments are illegal in Verilog-A.[39m
[31m│[39m[90m   ╭───── missing_assignment.va:12[39m
[31m│[90m12[0m [90m│[0m                 NumFingerDiff(nf, minSD, niIntD, nuEndD, nuIntS, nuEndS);
[31m│[0m   [90m┊[0m                 [91m━━━━━━┳━━━━━━
[31m│[0m   [90m┊[0m         [91mStatement without assignment[0m
[31m│[39m[90m   │[39m
[31m│[39m[90m   ╰───── [39m